File: HRMS.Core/Models/Searches/EmployeeSearch.cs
namespace HRMS.Core.Models.Searches;

public class EmployeeSearch : BaseSearch
{
    public string? Name { get; set; }

    public bool IncludeCity { get; set; }

    public bool IncludeCountry { get; set; }

    public bool IncludeEducation { get; set; }
}

File: HRMS.Desktop/lib/models/searches/employee_search.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search.dart';

part 'employee_search.g.dart';

@JsonSerializable()
class EmployeeSearch extends BaseSearch {
  String? name;
  bool includeCity = false;
  bool includeCountry = false;
  bool includeEducation = false;

  EmployeeSearch();

  factory EmployeeSearch.fromJson(Map<String, dynamic> json) =>
      _$EmployeeSearchFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$EmployeeSearchToJson(this);
}
